src/phpDocumentor/Parser/ServiceProvider.php

```diff
@@ -54,22 +54,27 @@ public function register(Application $app)
             );
         }
 
+        $app['parser.middleware.cache'] = $app->share(
```

Comment:
I upgraded pimple to v3 in #1904. Please rebase this and remove the call to $app->share. The new pimple shares services by default.